### PR TITLE
remove consts/inputs after use them as attr in frontend

### PR DESCRIPTION
**Pull request by fumihwh**
#70

### CODE REVIEW

**Comment by tjingrant** [comment]
I think this is a very good idea. But a safer way to do this is through dependency analysis; i.e. Analyzing in the end whether the input is still needed after we done converting. The reason is, what if TF decides to re-use some of the shared constants? You are assuming that all the consts have a reference count of exactly 1 (or less).

**Comment by fumihwh** [comment]
@tjingrant 
You are right.
We should first think about how to pass additional args to handler. #69 
Is `*args` good enough?

**Comment by tjingrant** [comment]
Hi, kwargs is fine and may be even better than what's done before.
However, I think your current implementation still has a problem: you'll remove constant (or input) tensors if they are used as an input to an operation not as an attribute to an operation; which is obviously quite problematic. I assume this is not the final version you want to merge?

**Comment by fumihwh** [comment]
@tjingrant 
No. Actually I remove consts which is used as attr.
Could you point out which one is `they are used as an input to an operation not as an attribute to an operation`?

**Comment by fumihwh** [comment]
And alternative way is create a `Protos` class to hold all protos and process by class method.

**Comment by tjingrant** [comment]
My bad, I could never remember if the filter condition is the condition to keep or to discard...

Let's examine the logic here, you are assuming:
`A tensor is used by an operation as an attribute -> they are safe to be deleted`

This is a bit unsettling. A very unlikely scenario (but perfectly legal) is when a user used a const tensor both as an attribute and as an input to an operation. I believe your code could cause trouble. Instead, I propose using this logic to determine which const tensor to drop:

A const tensor is not an input to any ONNX operation we generated -> it is safe to be dropped.

Besides, your implementation requires each handlers to record accurately which tensors are used; which is a whole other story..

What do you think?

**Comment by fumihwh** [comment]
`A const tensor is not an input to any ONNX operation we generated -> it is safe to be dropped.` sounds good.

So what we need to do is just loop `ops_proto` at last to find which const is not used as input and dropped it, right?

**Comment by tjingrant** [comment]
Yes, I believe so.

**Comment by fumihwh** [comment]
@tjingrant 
FIxed. Looks good now.

File: onnx_tf/frontend.py [hunk 1]
```diff
@@ -182,15 +190,16 @@ def _bin_op(cls, node, onnx_op):
...
 
   @classmethod
-  def handle_logical_or(cls, node, consts):
+  def handle_logical_or(cls, node, **kwargs):
     return cls._bin_op(node, "Or")
 
   @classmethod
-  def handle_pad(cls, node, consts):
+  def handle_pad(cls, node, **kwargs):
+    consts = kwargs["consts"]
```
Comment (tjingrant) [inline onnx_tf/frontend.py]: What's the purpose of this statement?

**Comment by tjingrant** [comment]
Looks good!

**Pull request merged**